contracts/facets/EmbalmerFacet.sol

```diff
@@ -34,267 +34,237 @@ contract EmbalmerFacet {
-            selectedArchaeologists.length
-        );
-
+        // create the sarcophagus
+        LibTypes.Sarcophagus storage sarcophagus = s.sarcophagi[sarcoId];
+        sarcophagus.name = sarcophagusParams.name;
+        sarcophagus.threshold = sarcophagusParams.threshold;
+        sarcophagus.resurrectionTime = sarcophagusParams.resurrectionTime;
+        sarcophagus.maximumRewrapInterval = sarcophagusParams.maximumRewrapInterval;
+        sarcophagus.arweaveTxIds = arweaveTxIds;
+        sarcophagus.embalmerAddress = msg.sender;
+        sarcophagus.recipientAddress = sarcophagusParams.recipientAddress;
+        sarcophagus.cursedArchaeologistAddresses = new address[](selectedArchaeologists.length);
+
+        // track total digging fees due upon creation of sarcophagus
         uint256 totalDiggingFees = 0;
 
         for (uint256 i = 0; i < selectedArchaeologists.length; i++) {
-            LibTypes.SelectedArchaeologistData memory arch = selectedArchaeologists[i];
-            LibUtils.revertIfArchProfileDoesNotExist(arch.archAddress);
-
-            // Confirm that the archaeologist list is unique. This is done by
-            // checking that the archaeologist does not already exist from
-            // previous iterations in this loop.
-            if (LibUtils.archaeologistExistsOnSarc(sarcoId, arch.archAddress)) {
-                revert LibErrors.ArchaeologistListNotUnique(
-                    cursedArchaeologists
-                );
+            // confirm archaeologist is registered
+            LibUtils.revertIfArchProfileDoesNotExist(selectedArchaeologists[i].archAddress);
+
+            // Confirm archaeologist isn't already cursed on sarcophagus
+            // todo: may be unnecessary, is cursing an archaeologist twice harming anybody but the caller?
+            if (
+                sarcophagus.cursedArchaeologists[selectedArchaeologists[i].archAddress].doubleHashedKeyShare != 0
+            ) {
+                revert LibErrors.ArchaeologistListNotUnique(selectedArchaeologists[i].archAddress);
             }
 
-            // Validate the archaeologist has signed off on the sarcophagus parameters: double hashed key share,
-            // arweaveTxId[1] (tx storing share on arweave), maximumRewrapInterval for sarcophagus
+            // Validate the archaeologist has signed off on the sarcophagus parameters
             LibUtils.verifyArchaeologistSignature(
-                arch.unencryptedShardDoubleHash,
+                selectedArchaeologists[i].doubleHashedKeyShare,
                 arweaveTxIds[1],
-                sarcophagus.maximumRewrapInterval,
-                sarcophagus.timestamp,
-                arch.diggingFee,
-                arch.v,
-                arch.r,
-                arch.s,
-                arch.archAddress
+                sarcophagusParams.maximumRewrapInterval,
+                sarcophagusParams.creationTime,
+                selectedArchaeologists[i].diggingFee,
```

Comment:
I'm interested in the difference in cost between having to do this "i-th" lookup every time vs simply accessing a known object in memory. Looks like this change was made because it's actually more efficient?

Reply:
Copying each of those structs from calldata into memory has some cost, so I'd rather just access it where it currently lives in calldata. I couldn't find any clear info on whether or not array indexing has extra cost though, I even asked on a forum after you posted this question because you made me curious. My feeling is that this is the more efficient way to do things, but I'd like to actually do a gas cost evaluation or see if we can get the yul code for this and figure out what's going on under the hood.